<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>850</width>
    <height>850</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="pipeline">
      <property name="sizeConstraint">
       <enum>QLayout::SetMinimumSize</enum>
      </property>
      <property name="leftMargin">
       <number>10</number>
      </property>
      <property name="topMargin">
       <number>10</number>
      </property>
      <property name="rightMargin">
       <number>10</number>
      </property>
      <property name="bottomMargin">
       <number>10</number>
      </property>
      <item>
       <spacer name="spacer1">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="buttonStart">
        <property name="text">
         <string>Choose image</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line1">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label11">
        <property name="text">
         <string>Create initial set of sample points</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formStep1">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item row="1" column="0">
         <widget class="QPushButton" name="backButton1">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Back</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="buttonSamplePoints">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Proceed</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label1">
          <property name="text">
           <string>Sample size:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="valueSampleSize">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label12">
        <property name="text">
         <string>Partition into Voronoï cells</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formStep2">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="QPushButton" name="backButton2">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Back</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="buttonPartitionVoronoi">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Proceed</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label13">
        <property name="text">
         <string>Add points</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formStep3">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label3">
          <property name="text">
           <string>Max diviation:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="valueMaxDiviation">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="backButton3">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Back</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="buttonAddAll">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Add all points</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="buttonAddPoint">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Add point</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="buttonUndoPoint">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Undo</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label14">
        <property name="text">
         <string>Uniformize the intensity</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formStep4">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="QPushButton" name="backButton4">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Back</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="buttonUniformizeIntensity">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Proceed</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label15">
        <property name="text">
         <string>Merge Voronoï cells</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formStep5">
        <item row="0" column="0">
         <widget class="QLabel" name="label5">
          <property name="text">
           <string>Min area:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="backButton5">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Back</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="buttonMergeCells">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Proceed</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="valueMinArea">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line6">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label16">
        <property name="text">
         <string>Create borders</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formStep6">
        <item row="0" column="0">
         <widget class="QLabel" name="label6">
          <property name="text">
           <string>Min width:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="valueMinWidth">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="backButton6">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Back</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="buttonCreateBorders">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Proceed</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line7">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label17">
        <property name="text">
         <string>Set density values</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formStep7">
        <item row="0" column="0">
         <widget class="QLabel" name="label7">
          <property name="text">
           <string>Budget:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="valueBudget">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="backButton7">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Back</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="buttonSetDensities">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Proceed</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line8">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonStop">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Finalize</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="spacer2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="image">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
